django/db/models/options.py

```diff
@@ -115,6 +115,7 @@ def __init__(self, meta, app_label=None):
         self.pk = None
         self.has_auto_field = False
         self.auto_field = None
+        self.has_readonly_fields = False
```

Comment:
Having a `@cached_property` for read-only fields would be more appropriate in this case. You'd just have to make sure you add `'readonly_fields'` to `FORWARD_PROPERTIES` to make sure the cache is busted on field addition through `add_field()`.